lib/authentication.js

```diff
@@ -0,0 +1,331 @@
+'use strict';
+
+const MAX_PASS_LENGTH = 128;
+const MIN_PASS_LENGTH = 10;
+const MIN_PHRASE_LENGTH = 20;
+
+const MIN_LOGIN_LENGTH = 6;
+const MAX_LOGIN_LENGTH = 255;
+
+const LOWERCASE_CHAR_CODES = {
+  charCodeRanges: [
```

Comment:
How did you generate those tables? It's better to have a script to do this, as for each new Unicode version this will have to change.